pkg/go/graph/weighted_graph_builder.go

```diff
@@ -18,60 +21,180 @@ func NewWeightedAuthorizationModelGraphBuilder() *WeightedAuthorizationModelGrap
 }
 
 func (wgb *WeightedAuthorizationModelGraphBuilder) Build(model *openfgav1.AuthorizationModel) (*WeightedAuthorizationModelGraph, error) {
-	g, err := NewAuthorizationModelGraph(model)
-	if err != nil {
-		return nil, err
+	wb := NewWeightedAuthorizationModelGraph()
+	// sort types by name to guarantee stable output
+	sortedTypeDefs := make([]*openfgav1.TypeDefinition, len(model.GetTypeDefinitions()))
+	copy(sortedTypeDefs, model.GetTypeDefinitions())
+
+	slices.SortFunc(sortedTypeDefs, func(a, b *openfgav1.TypeDefinition) int {
+		return cmp.Compare(a.GetType(), b.GetType())
+	})
+
+	for _, typeDef := range sortedTypeDefs {
+		wb.GetOrAddNode(typeDef.GetType(), typeDef.GetType(), SpecificType)
+
+		// sort relations by name to guarantee stable output
+		sortedRelations := make([]string, 0, len(typeDef.GetRelations()))
+		for relationName := range typeDef.GetRelations() {
+			sortedRelations = append(sortedRelations, relationName)
+		}
+
+		slices.Sort(sortedRelations)
+
+		for _, relation := range sortedRelations {
+			uniqueLabel := fmt.Sprintf("%s#%s", typeDef.GetType(), relation)
```

Comment:
performance: direct concatenation will be far faster than using `fmt.Sprintf`
```suggestion
			uniqueLabel := typeDef.GetType() + '#' + relation
```